### PR TITLE
Implement isEmpty predicate

### PR DESCRIPTION
**Pull request by Reachip**
This PR implement the "isEmpty" predicate.
Today, we are able to say:

```
GET https://api.foo.com
HTTP 200

[Asserts]
jsonpath "$.books" count == 0
```

We would like to be able to write:

```
GET https://api.foo.com
HTTP 200

[Asserts]
jsonpath "$.books" isEmpty
```

This predicate could work for collections or strings.

### CODE REVIEW

File: packages/hurl/src/runner/predicate.rs [hunk 1]
```diff
@@ -495,6 +496,28 @@ fn eval_something(
...
+                expected: 0.to_string(),
+                type_mismatch: false,
+            }),
+            Value::String(data) => Ok(AssertResult {
+                success: data.len() as i64 == 0,
+                actual: data.len().to_string(),
+                expected: 0.to_string(),
+                type_mismatch: false,
+            }),
+            _ => Ok(AssertResult {
```
Comment (fabricereix) [inline packages/hurl/src/runner/predicate.rs:514]: The `count` filter (https://github.com/Orange-OpenSource/hurl/blob/master/packages/hurl/src/runner/filter.rs#L117) can also take a `Bytes` or `Nodeset` value, we could therefore add them in the `empty` predicate.

**Comment by fabricereix** [comment]
@Reachip 
small detail, you could rebase all your commits into one?

**Comment by jcamiel** [comment]
I closed this PR and open another one

**Comment by jcamiel** [comment]
I'm closing this PR we'll reopen a new PR once the implementation is done.

**Pull request closed without merging**